### PR TITLE
Remove CURRENT_STATEMENT where not needed

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref https://github.com/rectorphp/rector/issues/7126

### CODE REVIEW

File: src/PhpParser/Node/NamedVariableFactory.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+final class NamedVariableFactory
+{
+    public function __construct(
+        private VariableNaming $variableNaming
+    ) {
+    }
+
+    public function createVariable(\PhpParser\Node $node, string $variableName): Variable
+    {
+        $currentStmt = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
```
Comment (samsonasik) [inline src/PhpParser/Node/NamedVariableFactory.php]: should can be replaced by `BetterNodeFinder->resolveCurrentStatement()` with set method modifier to public

https://github.com/rectorphp/rector-src/blob/80a04aa51f57c801cf30b81b3c02f3480471f278/src/PhpParser/Node/BetterNodeFinder.php#L541

Comment (TomasVotruba) [reply src/PhpParser/Node/NamedVariableFactory.php]: Even better, thanks :+1:

File: rules/DowngradePhp70/Rector/Spaceship/DowngradeSpaceshipRector.php [hunk 2]
```diff
@@ -94,17 +92,11 @@ public function refactor(Node $node): FuncCall
...
-        $currentStatement = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
-        if (! $currentStatement instanceof Node) {
-            throw new ShouldNotHappenException();
-        }
+        $variableAssignName = $this->namedVariableFactory->createVariable($node, 'battleShipcompare');
 
-        $scope = $currentStatement->getAttribute(AttributeKey::SCOPE);
-
-        $variableAssignName = $this->variableNaming->createCountedValueName('battleShipcompare', $scope);
         $variableAssign = new Variable($variableAssignName);
```
Comment (samsonasik) [inline rules/DowngradePhp70/Rector/Spaceship/DowngradeSpaceshipRector.php]: can directly assign  

```suggestion
        $variableAssign = $this->namedVariableFactory->createVariable($node, 'battleShipcompare');
```

Comment (TomasVotruba) [reply rules/DowngradePhp70/Rector/Spaceship/DowngradeSpaceshipRector.php]: Just noticed :+1:

**Pull request merged**